src/client/Client.js

```diff
@@ -365,6 +366,20 @@ class Client extends BaseClient {
       .then(data => new GuildPreview(this, data));
   }
 
+  /**
+   * Obtains the widget of a guild from Discord, available for all guilds.
+   * @param {GuildResolvable} guild The guild to fetch the widget for
+   * @returns {Promise<Widget>}
+   */
+  fetchWidget(guild) {
+    const id = this.guilds.resolveID(guild);
+    if (!id) throw new TypeError('INVALID_TYPE', 'guild', 'GuildResolvable');
+    return this.api
+      .guilds(id, 'widget.json')
+      .get()
+      .then(data => new Widget(this, data));
```

Comment:
I believe we've started to prefer async over then where possible, e.g.
```suggestion
  async fetchWidget(guild) {
    const id = this.guilds.resolveID(guild);
    if (!id) return Promise.reject(new TypeError('INVALID_TYPE', 'guild', 'GuildResolvable'));
    const data = await this.api.guilds(id, 'widget.json').get();
    return new Widget(this, data);
```